internal/slackdeps/http.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022-2025 Salesforce, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package slackdeps
+
+import (
+	"crypto/tls"
+	"net/http"
+	"time"
+)
+
+const (
+	// defaultTotalTimeout is the default timeout for the life of the request.
+	defaultTotalTimeout = 30 * time.Second
+)
+
+// HTTPClient interface for http.Client.
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+	Get(url string) (*http.Response, error)
+}
```

Comment:
note: I only added what we use today, but this will need to extend to `Post` and `Head` as we use it in other areas of the code base.